glotaran/analysis/problem.py

```diff
@@ -140,7 +146,7 @@ def parameters(self, parameters: ParameterGroup):
         self.reset()
 
     @property
-    def parameter_history(self) -> list[ParameterGroup]:
+    def parameter_history(self) -> ParameterHistory:
```

Comment:
This change from a list of ParameterGroup to a ParameterHistory introduced a bug in `_create_result` line 94-96 since calling len on the class is not supported:
```python
number_of_function_evaluation = (
        ls_result.nfev if ls_result is not None else len(problem.parameter_history)
    )
```

Could implement `__len__` in `ParameterHistory` or change the implementation in `_create_result`.

Reply:
Alternatively in _create_result the `number_records` property could be called but it might also be worth to do both.

```python
    @property
    def number_records(self) -> int:
        """Return the number of records in the history.

        Returns
        -------
        int
            The number of records.
        """
        return len(self._parameters)
```

Reply:
Resolved in 2edcca96b52d99162727aa1b18376d67ed068ca5 and 2edcca96b52d99162727aa1b18376d67ed068ca5